src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/Main.java

```diff
@@ -342,6 +342,9 @@ public String toString() {
                     }
                     break;
                 default:
+                    if (opt.startsWith("-agentlib:jdwp")) {
+                        javacOpts.add("-g");
+                    }
```

Comment:
The JDWP agent does expose an agent property for tools for the server=y case but it wouldn't be appropriate for the source launcher to look for that. So checking for a CLI option is probably okay, just needs to be  "-agentlib:jdwp:" to avoid matching other agents prefixed with "jdwp". Also check for "-Xrunjdwp:" too because -Xrun is still used.